thirdparty/hnswlib/hnswlib/hnswalg.h

```diff
@@ -1028,6 +1030,24 @@ class HierarchicalNSW : public AlgorithmInterface<dist_t> {
         if (cur_element_count == 0)
             return {};
 
+        if (!bitset.empty() && bitset.count() > (cur_element_count * kHnswBruteForceFilterRate)) {
+            assert(cur_element_count == bitset.size());
+            knowhere::ResultMaxHeap<dist_t, labeltype> max_heap(k);
+            for (labeltype id = 0; id < cur_element_count; ++id) {
+                if (!bitset.test(id)) {
+                    dist_t dist = fstdistfunc_(query_data, getDataByInternalId(id), dist_func_param_);
+                    max_heap.Push(dist, id);
+                }
+            }
+            const size_t len = std::min(max_heap.Size(), k);
```

Comment:
for Knowhere we return -1 if the heap size is smaller than k

Reply:
Plz double check the non-brute force algo to confirm

Reply:
same with `searchKnn` https://github.com/milvus-io/knowhere/blob/7a93cb6fd46ba315049b4d6e9cd5248d52c8fb52/thirdparty/hnswlib/hnswlib/hnswalg.h#L1102, handled outside https://github.com/milvus-io/knowhere/blob/7a93cb6fd46ba315049b4d6e9cd5248d52c8fb52/src/index/hnsw/hnsw.cc#L154